subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -532,4 +533,28 @@ public void setArchiveBase(PathBase archiveBase) {
     public Property<Integer> getNetworkTimeout() {
         return networkTimeout;
     }
+
+    /**
+     * Sets if the task will validate the distribution url that has been configured.
+     *
+     * @since 8.2
+     */
+    @Incubating
+    @Option(option = "validate-url", description = "Sets task to validate the configured distribution url")
+    public void setValidateDistributionUrl(Boolean validateDistributionUrl) {
+        this.validateDistributionUrl.set(validateDistributionUrl);
+    }
+
+    /**
+     * Indicates if this task will validate the distribution url that has been configured.
+     *
+     * @since 8.2
+     * @return whether this task will validate the distribution url
+     */
+    @Incubating
+    @Input
+    @Optional
+    public Property<Boolean> getValidateDistributionUrl() {
```

Comment:
When adding a new public property, you also need to update `subprojects/docs/src/docs/dsl/org.gradle.api.tasks.wrapper.Wrapper.xml`

You just need to add something like:
```
            <tr>
                <td>validateDistributionUrl</td>
                <td><literal>true</literal></td>
            </tr>
```